code-challenges.test.js

```diff
@@ -61,3 +119,18 @@ var numbersToAdd3 = []
 
 
 // b) Create the function that makes the test pass.
+
+describe("The sumAll function.",()=>{
+    test("takes in an array and returns a new array with the sum of all numbers in old array", () => {
+        expect(sumAll(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(sumAll(numbersToAdd2)).toEqual([0, 7, -1, 11]            )
+        expect(sumAll(numbersToAdd3)).toEqual([])
+
+    })
+})
+
+const sumAll = (arr) =>{
+    let sumdArr = arr.map((sum => value => sum += value)(0))
+    return sumdArr
+}
+console.log(sumAll(numbersToAdd1))
```

Comment:
So yea this blew my mind. You are using a syntax called currying: In mathematics and computer science, currying is the technique of converting a function that takes multiple arguments into a sequence of functions that each takes a single argument. (Wikipedia) This is the first time I have seen it so I'll look into it a bit more.